### ISSUE TITLE
Provide an efficient logits mask index

### ISSUE THREAD

**Issue by Dan-wanna-M** [Dan-wanna-M/kbnf#16]
Essentially [this](https://github.com/vllm-project/vllm/issues/3567). Expose a reference to the allowed_token_ids would be enough on the rust side and we probably need to get a &[u8] that automatically converts to PyBytes for Python FFI.

**Comment by Dan-wanna-M** [comment]
implemented in `0.3.5-python`
